src/context/AppProvider/AppContext.tsx

```diff
@@ -289,31 +297,44 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
   /**
    * fetch forex spot and history for user's selected currency
    */
-  const selectedCurrency = useAppSelector(state => selectSelectedCurrency(state))
+  const symbol = useAppSelector(state => selectSelectedCurrency(state))
+  const timeframe = DEFAULT_HISTORY_TIMEFRAME
+  const priceHistoryArgs = useMemo(() => ({ symbol, timeframe }), [symbol, timeframe])
+  const { error: priceHistoryError } = useFindPriceHistoryByFiatSymbolQuery(priceHistoryArgs)
```

Comment:
using these fancy autogenerated hooks that expose errors instead of manually dispatching